packages/issuer/lib/VcIssuerBuilder.ts

```diff
@@ -0,0 +1,78 @@
+import { ICredentialIssuerMetadataParametersV1_11, ICredentialSupportedV1_11, IIssuerDisplay, invalid_request } from '@sphereon/openid4vci-common'
+
+import { VcIssuer } from './VcIssuer'
+
+export class VcIssuerBuilder {
+  credentialIssuer?: string
+  authorizationServer?: string
+  credentialEndpoint?: string
+  batchCredentialEndpoint?: string
+  issuerDisplay?: IIssuerDisplay[]
+  credentialsSupported?: ICredentialSupportedV1_11[]
+  userPinRequired?: boolean
+
+  public withCredentialIssuer(issuer: string): VcIssuerBuilder {
+    this.credentialIssuer = issuer
+    return this
+  }
+  public withAuthorizationServer(authorizationServer: string): VcIssuerBuilder {
+    this.authorizationServer = authorizationServer
+    return this
+  }
+
+  public withCredentialEndpoint(credentialEndpoint: string): VcIssuerBuilder {
+    this.credentialEndpoint = credentialEndpoint
+    return this
+  }
+
+  public withBatchCredentialEndpoint(batchCredentialEndpoint: string): VcIssuerBuilder {
+    this.batchCredentialEndpoint = batchCredentialEndpoint
+    return this
+  }
+
+  public withIssuerDisplay(issuerDisplay: IIssuerDisplay | IIssuerDisplay[]): VcIssuerBuilder {
+    if (!Array.isArray(issuerDisplay)) this.issuerDisplay = this.issuerDisplay ? [...this.issuerDisplay, issuerDisplay] : [issuerDisplay]
```

Comment:
If isseurDisplay is not an array and this.issuerDisplay is not present you return issuerDisplay as an array? Isn't it the other way around?

Reply:
Ok!